app/models/template.rb

```diff
@@ -73,8 +73,99 @@ def ignore_default
     self
   end
 
+  # Set attributes based on template +text and metadata found in this +text
+  # the metadata parsing can be adjusted using +options
+  def import_without_save(text, options = {})
+    self.template = text
+    @importing_metadata = self.class.parse_metadata(text)
+    Foreman::Logging.logger('app').debug "setting attributes for #{self.name} with id: #{self.id || 'N/A'}"
```

Comment:
This line (logger) and mixture of class and instance methods is a good sign that we are abusing the model class too much. We have a God object problem here, please pick any reasonable Ruby design pattern and extract import logic into separate class responsible only for importing. This is my biggest concern so far.

Reply:
Sorry, I don't understand how using Logger contributes to god object problem. Logger is singleton. Normally you'd write logger.debug but here you're out of that scope. Parsing metadata is not depending on instance, which is why it's class method. Handling specific attributes is. I could extract importing to separate service object, but this logic seems very related to this model.

If I extract this to separate object, we'd also lose the inheritance that extends the import logic specific to children. See the ptable and provisioning template only extends it with logic specific for their classes. I can imagine copying this concept to service objects too but IMHO that does not bring value and makes the code less readable. 200 lines model does not seem as god object, especially if half of that are private methods.

Reply:
The problem is that importing mechanism should not be that tightly coupled with the model. It is definitely not the biggest God object we have for sure, but I think this patch is a good candidate for Builder design pattern for example as one day we might want to import templates from different sources. I will not insist, but consider that for another moment.

Reply:
I agree that additional logic, such as getting templates data from somewhere should be kept out of this model, and if you look at foreman_templates we have separate service objects that does parsing git/fs and then use the `import` method from this model. Import should only understand attributes that this model has. Perhaps the metadata parsing could be separated later, but long term I'd like to make it new attribute of template so we don't need to keep it in template content. That's out of scope for this PR though.